mantle/kola/tests/ignition/qemufailure.go

```diff
@@ -101,3 +129,206 @@ func ignitionFailure(c cluster.TestCluster) error {
+	builder := platform.NewQemuBuilder()
+
+	consoleFile, err := builder.TempFile("console.log")
+	if err != nil {
+		return err
+	}
+	builder.ConsoleFile = consoleFile.Name()
+
+	// get current path and create tmp dir
+	fakeBootFile, err := builder.TempFile("fakeBoot")
+	if err != nil {
+		return err
+	}
+
+	// Truncate the file to 1 gigabyte
+	const oneGB = 1 << 30
+	err = fakeBootFile.Truncate(oneGB)
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.Command("mkfs.ext4", "-L", "boot", fakeBootFile.Name())
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		c.Fatal(err)
+	}
+
+	err = builder.AddBootDisk(&platform.Disk{
+		BackingFile: kola.QEMUOptions.DiskImage,
+	})
+	if err != nil {
+		return err
+	}
+	err = builder.AddDisk(&platform.Disk{
+		BackingFile:   fakeBootFile.Name(),
+		BackingFormat: "raw",
+	})
+	if err != nil {
+		return err
+	}
+	builder.MemoryMiB = 1024
+	builder.Firmware = kola.QEMUOptions.Firmware
+	inst, err := builder.Exec()
+	if err != nil {
+		return err
+	}	// platform.Manhole()
+	defer inst.Destroy()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	errchan := make(chan error)
+	go func() {
+		resultingError := inst.WaitAll(ctx)
+
+		if resultingError == nil {
+			resultingError = fmt.Errorf("bootfs unexpectedly succeeded")
+		} else if resultingError == platform.ErrInitramfsEmergency {
+			// Expectred initramfs failure, checking the console file to insure
+			// that coreos-unique-boot.service failed
+			b, err := os.ReadFile(builder.ConsoleFile)
+			if err != nil {
+				panic(err)
+			}
+			isExist, err := regexp.Match("Error: System has 2 devices with a filesystem labeled 'boot'", b)
+			if err != nil {
+				panic(err)
+			}
+			if isExist {
+				// The expected case
+				resultingError = nil
+			} else {
+				resultingError = errors.Wrapf(err, "expected coreos-unique-boot.service to fail")
+			}
+		} else {
+			resultingError = errors.Wrapf(err, "expected initramfs emergency.target error")
+		}
+		errchan <- resultingError
+	}()
+
+	select {
+	case <-ctx.Done():
+		if err := inst.Kill(); err != nil {
+			return errors.Wrapf(err, "failed to kill the vm instance")
+		}
+		return errors.Wrapf(ctx.Err(), "timed out waiting for initramfs error")
+	case err := <-errchan:
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+// Use ignition config to create a second bootfs
+// 1 - produce an ignition file that format a disk with a"boot" label.
+// 2 - boot the VM with the ignition file and an extra disk.
+// 3 - observe the failure 
+func dualBootfsIgnitionFailure(c cluster.TestCluster) error {
+	builder := platform.NewQemuBuilder()
+	// inserting log file for troubleshooting
+	// file located in coreos-assembler directory
+
+	// consoleFile, err := builder.TempFile("console.log")
+	// if err != nil {
+	// 	return err
+	// }
+	//builder.ConsoleFile = consoleFile.Name()
+	builder.ConsoleFile = "console.txt"
```

Comment:
Leftover from testing? I think we want the `builder.TempFile()` version, right?

Reply:
Oops!